lib/python-beta/src/bailo/helper/model.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+import os
+from typing import Any
+
+from bailo.core import Client, ModelVisibility
+from bailo.helper.utils import retrieve_model
+
+
+class Model:
+    def __init__(
+        self,
+        client: Client,
+        name: str,
+        description: str,
+        visibility: ModelVisibility | None = ModelVisibility.Public,
+        model_card: dict[str, Any] | None = None
+    ):
+        self.client = client
+        self.name, self.description, self.visibility = name, description, visibility
+        self.model_id = None
+        self.model_version = None
+        self.model_card_version = None
+        self.model_card_schema = 'minimal-general-v10-beta'
+        self.__model_card_original = model_card
+        self.__model_card = model_card
+        self.__local_dir = f"./temp_{name.strip(' ')}"  # property tbc
+        self.__mc_change = 0
```

Comment:
I'm not a fan of having these private methods to track whether properties have changed or not.  Can we just make everything idempotent?  

E.g. when a user runs `add_file()` it _immediately_ updates the release with the new file.  It means we don't have to track internal state.